File: app/src/main/java/com/wojdor/hearthstonecards/application/util/FileStorage.java
package com.wojdor.hearthstonecards.application.util;

import android.content.Context;
import android.graphics.Bitmap;

import java.io.File;
import java.io.FileOutputStream;

import timber.log.Timber;

public class FileStorage {

    private static final String PATH_FORMAT = "%s/%s";
    private Context context;

    public FileStorage(Context context) {
        this.context = context;
    }

    public void write(String fileName, Bitmap bitmap) {
        FileOutputStream outputStream;
        try {
            outputStream = context.openFileOutput(fileName, Context.MODE_PRIVATE);
            bitmap.compress(Bitmap.CompressFormat.PNG, 50, outputStream);
            outputStream.close();
        } catch (Exception error) {
            Timber.e(error);
        }
    }

    public File get(String fileName) {
        String path = String.format(PATH_FORMAT, context.getFilesDir().getAbsolutePath(), fileName);
        return new File(path);
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/application/util/FileStorage.kt
package com.wojdor.hearthstonecards.application.util

import android.content.Context
import android.graphics.Bitmap
import timber.log.Timber
import java.io.File
import java.io.FileOutputStream

class FileStorage(private val context: Context) {

    fun write(fileName: String, bitmap: Bitmap) {
        val outputStream: FileOutputStream
        try {
            outputStream = context.openFileOutput(fileName, Context.MODE_PRIVATE)
            bitmap.compress(Bitmap.CompressFormat.PNG, 50, outputStream)
            outputStream.close()
        } catch (error: Exception) {
            Timber.e(error)
        }
    }

    fun get(fileName: String): File {
        val path = String.format(PATH_FORMAT, context.filesDir.absolutePath, fileName)
        return File(path)
    }

    companion object {
        private const val PATH_FORMAT = "%s/%s"
    }
}
